### ISSUE TITLE
meta-chip layers

### ISSUE THREAD

**Issue by soderstrom-rikard** [agherzan/meta-chip#8]
Hi Andrei,

Noticed your answer in meta-sunxi (I haven't been around much this year).
Seems like there are a number of meta-chip layers out there now.

1. Yours (the one referenced from openembedded layers index)
2. chip-community/meta-chip@b86d7ba142fda299cc93d25520ad45d8702fb54b (fork of myfreescalewebpage/meta-chip@b86d7ba142fda299cc93d25520ad45d8702fb54b)
3. My old PR linux-sunxi/meta-sunxi#126 (which I closed today)

Maybe we should find common ground and just have one we all work together on?
The one currently in chip-community seems to be most feature-rich (most image variants), whilst yours seems to be most minimalistic (best ground to build from).

I also invited you as a member to chip-community, hope you don't mind.

Cheerio
Soderstrom

**Comment by agherzan** [comment]
Somehow this ended up in spam. Sorry about it. We should definitely find a common ground to it. Let's bring all the pieces together and consolidate a layer. What do you think?

**Comment by soderstrom-rikard** [comment]
Yes. We should probably start by analysing the different layers. I may have some time to start looking at it this weekend. If so, I will give my initial impressions here. @myfreescalewebpage may be interested if we ask. There is also a third layer now (for new nands and CHIP pro) https://github.com/texierp/meta-chip

Cheerio
Soderstrom

**Comment by myfreescalewebpage** [comment]
Hi @agherzan  and @soderstrom-rikard 

Great idea. I was also working on CHIP pro but not totally finished because missing some time. I have started with uboot official repo which is interesting to follow easily future evolutions, but requires a little bit more job. My progress is : build seems ok, but difficulties to flash the board.

https://github.com/texierp/meta-chip (hummm seems to be a fork of yours @agherzan, not very fairplay ....) looks great expect the readme..... I have my personal point of view but it looks too poor to easily start with the layer. Hope it is working anyway, I can test it asap if you want ? Just afraid CHIP SDK must be fully installed.... but maybe then we can extract only a part of the CHIP SDK and integrate it to CHIP tools.... will see.

Joel

**Comment by texierp** [comment]
Hi @soderstrom-rikard, @agherzan and @myfreescalewebpage 

Indeed, my layer is just a fork of this repo, with some improvements (new nands and C.H.I.P Pro support -> tested by Stefano Cordibella : no C.H.I.P Pro from my side).

@myfreescalewebpage you're right, CHIP SDK is mandatory. Rather than extract something, maybe we can used the repo tool (like FSL BSP) and create a specific fork of CHIP SDK (yocto branch ?).

But who maintain this ? :)

BR 

PJ

**Comment by agherzan** [comment]
I'm going to fix the repo that I have (has some build issues). I triggered a couple of builds. If you guys want, I am OK to consolidate all the change in my repository and I will try to maintain it from now on. Didn't have time until. How does it sound?

**Comment by agherzan** [comment]
BTW I'm thinking of setting up a configuration similar to what we did in meta-rpi (https://github.com/agherzan/meta-raspberrypi):
1. documentation
2. jenkins integration with PR builds

**Comment by belzerus** [comment]
@agherzan sounds like an good idea to have this in the same way as for meta-raspberrypi. Will you send out some info on the yocto mailing list about this when/if going for the jenkins integration for meta-chip as well?

**Comment by belzerus** [comment]
@agherzan @soderstrom-rikard and @myfreescalewebpage have you come to any conclusions about what layer to use? At least my feeling is that we have a good enough ground to start from using this meta-chip layer, together with the fact that there have been some increased activity in this layer, through the Yocto Project mailing list lately. So this feels like the layer with most active contributors at this time. But if going for this layer I guess it should be synchronized in some way with the chip-community.

**Comment by agherzan** [comment]
Just sent an email to the ml @belzerus

**Comment by soderstrom-rikard** [comment]
@agherzan  @myfreescalewebpage @belzerus 
I agree we should go forth with this one.

For now, I will rename the one in chip-community and add this one too.

Not only because this has most contributors, but also because it has the least dependencies to other layers.
